libraries/core_libs/consensus/src/final_chain/final_chain.cpp

```diff
@@ -89,23 +90,48 @@ class FinalChainImpl final : public FinalChain {
   std::shared_ptr<FinalizationResult> finalize_(SyncBlock&& new_blk, std::vector<h256>&& finalized_dag_blk_hashes,
                                                 finalize_precommit_ext const& precommit_ext) {
     auto batch = db_->createWriteBatch();
-    SharedTransactions to_execute;
-    {
-      // This artificial scope will make sure we clean up the big chunk of memory allocated for this batch-processing
-      // stuff as soon as possible
-      to_execute.reserve(new_blk.transactions.size());
-      for (size_t i = 0; i < new_blk.transactions.size(); ++i) {
-        auto& trx = new_blk.transactions[i];
-        if (!(replay_protection_service_ &&
-              replay_protection_service_->is_nonce_stale(trx->getSender(), trx->getNonce()))) [[likely]] {
-          to_execute.push_back(std::move(trx));
-        }
+
+    // Transactions to be executed in evm
+    SharedTransactions txs_to_execute;
+    txs_to_execute.reserve(new_blk.transactions.size());
+
+    // Dag blocks validators that included transactions to be executed as first in their blocks
+    std::vector<addr_t> txs_to_execute_validators;
+    txs_to_execute_validators.reserve(new_blk.transactions.size());
+
+    // Creates rewards stats
+    RewardsStats rewards_stats;
+    rewards_stats.processStats(new_blk);
```

Comment:
maybe I am wrong but `txs_to_execute_validators` could be returned by this function -> `processStats` and txs_to_execute we will not need as we can use `new_blk.transactions`